corehq/tabs/uitab.py

```diff
@@ -8,6 +8,18 @@
 from django.conf import settings
 
 
+def url_is_location_safe(url):
+    from corehq.apps.users.location_access_restrictions import is_location_safe
+    try:
+        match = resolve(url)
+    except Resolver404:
+        return False
+    else:
+        if is_location_safe(match.func, match.args, match.kwargs):
+            return True
+        return False
```

Comment:
This is the heart of the UITab changes - basically it filters the dropdown and sidebar lists by the URL of each link.  That feels like a lot of repetitive work that should be optimizable, but not sure how best to do that, and I believe these bars are cached anyways.